js/source/worker_tile.js

```diff
@@ -98,10 +104,15 @@ WorkerTile.prototype.parse = function(data, layers, actor, callback) {
         symbolBuckets = this.symbolBuckets = [],
         otherBuckets = [];
 
+    featureTree.numberToLayerIDs = [];
+
     for (var id in bucketsById) {
         bucket = bucketsById[id];
         if (bucket.features.length === 0) continue;
 
+        bucket.index = featureTree.numberToLayerIDs.length;
+        featureTree.numberToLayerIDs.push(bucket.layerIDs);
```

Comment:
Could we get rid of `numberToLayerIDs` by changing `buckets` to an object indexed by `bucket.id`? If so, that'd be a cleaner solution and relatively easy change.